ls1intum/Artemis
# `Programming exercises`: Improve gradle templates for Java programming exercises

### Checklist
#### General
#### Server
#### Changes affecting Programming Exercises
- [x] I tested **all** changes and their related features with **all** corresponding user types on Test Server 2 (Jenkins and Gitlab) (does not apply, Gradle is not supported for this setup yet)

### Motivation and Context
This PR improves the Gradle template files for Java programming exercises and reduces the number of used configuration files.

### Description
- Upgrade to the newest Gradle Wrapper version 7.4.2. Adaption to the artemis-maven-template has been done
- Move required configuration from settings.gradle and gradle.properties to build.gradle
- Delete settings.gradle and gradle.properties

### Steps for Testing
Prerequisites:
- 1 Instructor
- Testserver with Atlassian Setup (TS0, TS1, TS5)

1. Log in to Artemis
2. Navigate to Course Administration
3. Create new Programming Exercises for the combinations 'Plain Java - Gradle' and 'Gradle - Gradle' each with SCA enabled and disabled.
4. Verify that the initial result is created correctly.
5. Import the project from the instructor context into your local IDE (use 'Import as Gradle Project') and verify that everything works as expected.
6. Start the exercise as a student, open in your local IDE and verify that everything works as expected.
(Committing and pushing are optional steps, nothing should have changed here because this PR is unrelated to this and related functionality). 

### Review Progress
#### Code Review
#### Manual Tests
